apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -159,9 +166,16 @@ export class OAuthService {
 			oauthResponse.jwt = jwtResponse;
 			return oauthResponse;
 		} catch (error) {
-			this.logger.log(error);
-			const system: System = await this.systemRepo.findById(systemId);
-			return this.getOAuthError(error as string, system.oauthConfig?.provider as string);
+			const oauthResponse: OAuthResponse = await this.systemRepo
+				.findById(systemId)
+				.then((system: System) => {
```

Comment:
Can you explain the benefit of an await in this case

Reply:
In generall we should avoid mixing of await and then chains. This are 2 different notations. The first use try catch notation for error handling, the other the .catch() with return a promise rejection. 
It is hard to read if both are mixing.

For us, we have define that we only want to use await in future, because it is easer to add changes and it is more readable.